DOCS-2541: Move component APIs (#3504)
viamrobotics/docs: docs/appendix/apis/components/arm.md

---
title: "Arm API"
linkTitle: "Arm"
weight: 20
type: "docs"

> description: "The arm API allows you to give commands to your arm components for linear motion planning, including self-collision prevention and obstacle avoidance."

Reviewer: The first half of each component description is redundant with the "<component> API" title. Can be much less verbose.
```suggestion
description: "Give commands to your arm components for linear motion planning, including self-collision prevention and obstacle avoidance."
```
or better yet
```suggestion
description: "Commands for linear motion planning, including self-collision prevention and obstacle avoidance."
```

icon: true
images: ["/icons/components/arm.svg"]
---

> The arm API allows you to give commands to your [arm components](/components/arm/) for linear motion planning, including self-collision prevention and obstacle avoidance.

Author: I see what you mean but you could have many arm components and I feel like singular feels weird. Let's leave it as is for now, we can always make more changes

---
title: "Arm API"
linkTitle: "Arm"
weight: 20
type: "docs"

> description: "The arm API allows you to give commands to your arm components with code instead of with the graphical interface of the Viam app"

Reviewer: This... feels odd. It seems like it doesn't need to be compared to the GUI.
```suggestion
description: "The arm API allows you to programmatically control your arm components."
```
or something

icon: true
images: ["/icons/components/arm.svg"]
---

> The arm API allows you to give commands to your [arm components](/components/arm/) with code instead of with the graphical interface of the [Viam app](https://app.viam.com/).

Reviewer: As above, this feels almost like a distraction. Like, if I'm a dev, maybe I came here and never would've even considered using the GUI for prod use case--I just need an API just as I need an API for any other thing in the world.

The arm component supports the following methods:

{{< readfile "/static/include/components/apis/generated/arm-table.md" >}}

## Establish a connection

To get started using Viam's SDKs to connect to and control your machine, go to your machine's page on [the Viam app](https://app.viam.com), navigate to the **CONNECT** tab's **Code sample** page, select your preferred programming language, and copy the sample code.

{{% snippet "show-secret.md" %}}

> When executed, this sample code will create a connection to your machine as a client.

Reviewer: ```suggestion
When executed, this sample code creates a connection to your machine as a client.
```
[nit]

> Then control your machine programmatically by adding API method calls as shown in the following examples.

Reviewer: This feels a bit unnecessary, like if someone knows what an API is, they don't need this sentence.

Author: ```suggestion
```

> These examples assume you have an arm called `"my_arm"` configured as a component of your machine.

Reviewer: ```suggestion
The examples below assume you have an arm called `"my_arm"` configured as a component of your machine.
```
(if sentence above gets removed)
